timemachine/cpp/src/context.cu

```diff
@@ -61,46 +63,66 @@ void Context::add_observable(Observable *obs) {
     this->observables_.push_back(obs);
 }
 
-std::vector<double> Context::multiple_steps(
+std::array<std::vector<double>, 2> Context::multiple_steps(
     const std::vector<double> &lambda_schedule,
-    int store_du_dl_freq) {
+    int store_du_dl_freq,
+    int store_x_freq) {
     unsigned long long *d_du_dl_buffer = nullptr;
     // try catch block is to deal with leaks in d_du_dl_buffer
-    if(store_du_dl_freq==0) {
-        store_du_dl_freq = lambda_schedule.size();
+    if(store_du_dl_freq <= 0) {
+        throw std::runtime_error("store_du_dl_freq <= 0");
+    }
+    if(store_x_freq <= 0) {
+        throw std::runtime_error("store_x_freq <= 0");
```

Comment:
Should we also catch here if the user requests an x buffer that would be too big? (say if len(lambda_schedule) = 1e7 and store_x_freq = 1)

Reply:
The limits are going to be GPU memory size dependent. These types of checks are best left on the application code side of things (there's lots of other ways to shoot one self in the foot). The default behavior here is not storing any frames, so I think we're pretty safe.